beacon-chain/blockchain/execution_engine.go

```diff
@@ -279,12 +280,20 @@ func (s *Service) pruneInvalidBlock(ctx context.Context, root, parentRoot, lvh [
 func (s *Service) getPayloadAttribute(ctx context.Context, st state.BeaconState, slot primitives.Slot, headRoot []byte) (bool, payloadattribute.Attributer) {
 	emptyAttri := payloadattribute.EmptyWithVersion(st.Version())
 
-	val, ok := s.trackedProposer(st, slot)
-	if !ok && !features.Get().PrepareAllPayloads {
-		return false, emptyAttri
+	// If it is an epoch boundary then process slots to get the right
+	// shuffling before checking if the proposer is tracked. Otherwise
+	// perform this check before. This is cheap as the NSC has already been updated.
+	var val cache.TrackedValidator
```

Comment:
When using `PrepareAllPayloads`, the fee recipient of the validator at index 0 is always used as the default value for the sending attribute. Is that acceptable?

Reply:
Ignore what I said. I think it's okay. It won't be `nil` as we return an empty `TrackedValidator`, the fee recipient would be zero and burned.